flask_pydantic_spec/spec.py

```diff
@@ -62,16 +65,58 @@ def __init__(
         self.models: Dict[str, Any] = {}
         if app:
             self.register(app)
+        if blueprint:
+            self.register(blueprint)
 
-    def register(self, app: Flask) -> None:
+    def register(
+        self,
+        app_or_blueprint: Union[Flask, Blueprint],
+        register_route: bool = True,
+    ) -> None:
         """
         register to backend application
 
         This will be automatically triggered if the app is passed into the
         init step.
         """
-        self.app = app
-        self.backend.register_route(self.app)
+        if isinstance(app_or_blueprint, Flask):
+            self.app = app_or_blueprint
+            self.backend.app = app_or_blueprint
+            if register_route:
+                self.register_spec_routes(self.app)
+
+        elif isinstance(app_or_blueprint, Blueprint):
+            self.blueprint = app_or_blueprint
+            if register_route:
+                self.register_spec_routes(self.blueprint)
+        else:
+            raise TypeError(f"unknown type provided {app_or_blueprint.__class__}")
```

Comment:
The register routes part could be moved outside the condition.

```suggestion
        if isinstance(app_or_blueprint, Flask):
            self.app = app_or_blueprint
            self.backend.app = app_or_blueprint
        elif isinstance(app_or_blueprint, Blueprint):
            self.blueprint = app_or_blueprint
        else:
            raise TypeError(f"unknown type provided {app_or_blueprint.__class__}")
        
        if register_route:
            self.register_spec_routes(self.app)
```